### PR TITLE
[2775] feat(core): Add JDBC backend operations for topic

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

Add JDBC backend operations for topic

### Why are the changes needed?

Implement support for storing messaging catalog metadata in the JDBC backend.
Fix: #2775 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

tests added

### CODE REVIEW

**Comment by mchades** [comment]
@YxAc Could you please help to review when you have time, thx~

**Comment by jerryshao** [comment]
@xloya would you please help to review this PR?

File: scripts/mysql/schema-0.5.0-mysql.sql [hunk 1]
```diff
@@ -98,4 +98,22 @@ CREATE TABLE IF NOT EXISTS `fileset_version_info` (
...
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin COMMENT 'fileset version info';
+
+CREATE TABLE IF NOT EXISTS `topic_meta` (
+    `topic_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'topic id',
+    `topic_name` VARCHAR(128) NOT NULL COMMENT 'topic name',
+    `metalake_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'metalake id',
+    `catalog_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'catalog id',
+    `schema_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'schema id',
+    `comment` VARCHAR(256) DEFAULT '' COMMENT 'topic comment',
+    `properties` MEDIUMTEXT DEFAULT NULL COMMENT 'topic properties',
```
Comment (xloya) [inline scripts/mysql/schema-0.5.0-mysql.sql:110]: Will there be multiple version requirements for this property in the future?

File: core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/TopicMetaMapper.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+          + " #{topicMeta.currentVersion},"
+          + " #{topicMeta.lastVersion},"
+          + " #{topicMeta.deletedAt}"
+          + " )")
+  void insertTopicMeta(@Param("topicMeta") TopicPO topicPO);
+
+  @Insert(
+      "INSERT INTO "
+          + TABLE_NAME
+          + "(topic_id, topic_name, metalake_id,  catalog_id, schema_id,"
```
Comment (xloya) [inline core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/TopicMetaMapper.java:41]: unnecessary spaces before `catalog_id`.

File: core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/TopicMetaMapper.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+          + " comment, properties, audit_info,  current_version, last_version,"
```
Comment (xloya) [inline core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/TopicMetaMapper.java:42]: Ditto.

File: core/src/main/java/com/datastrato/gravitino/storage/relational/service/TopicMetaService.java [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        case 2:
+          parentEntityId =
+              SchemaMetaService.getInstance().getSchemaIdByCatalogIdAndName(parentEntityId, name);
+          builder.withSchemaId(parentEntityId);
+          break;
+      }
+    }
+  }
+
+  public Object getTopicByIdentifier(NameIdentifier identifier) {
```
Comment (xloya) [inline core/src/main/java/com/datastrato/gravitino/storage/relational/service/TopicMetaService.java:152]: Why use `Object`?

File: core/src/main/java/com/datastrato/gravitino/storage/relational/po/TopicPO.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    if (!(o instanceof TopicPO)) {
+      return false;
+    }
+    TopicPO topicPO = (TopicPO) o;
+    return Objects.equals(topicId, topicPO.topicId)
+        && Objects.equals(topicName, topicPO.topicName)
+        && Objects.equals(metalakeId, topicPO.metalakeId)
+        && Objects.equals(catalogId, topicPO.catalogId)
+        && Objects.equals(schemaId, topicPO.schemaId)
+        && Objects.equals(auditInfo, topicPO.auditInfo)
```
Comment (xloya) [inline core/src/main/java/com/datastrato/gravitino/storage/relational/po/TopicPO.java:43]: It seems that the `properties` field is not added to the equals / hashCode computing.

File: scripts/mysql/schema-0.5.0-mysql.sql (hunk 1 above)
Comment (mchades) [reply scripts/mysql/schema-0.5.0-mysql.sql:110]: In my opinion, properties do not need multiple versions, while the schema of messages may require multiple versions, but we do not support message schema at this version.

**Review by jerryshao — APPROVED** [review]
LGTM.

**Pull request merged**